chapter-05/final/repository/envelope.go

```diff
@@ -0,0 +1,99 @@
+package repository
+
+import (
+	"errors"
+	"github.com/ablqk/tiny-go-projects/chapter-05/layered/money"
+)
+
+const baseCurrencyCode = "EUR"
+
+type Envelope struct {
+	Cube EnvelopeCube `xml:"Cube"`
```

Comment:
xml package supports `a>b>c` notation. You can use:

```
type EurofXref struct {
	Rates []CurrencyRate `xml:"Cube>Cube>Cube"`
}
type CurrencyRate struct {
	Currency string  `xml:"currency,attr"`
	Rate     float32 `xml:"rate,attr"`
}
```

Reply:
🧊 Thanks for the insight!